multiversx_sdk/abi/serializer.py

```diff
@@ -108,6 +116,20 @@ def _deserialize_variadic_values(self, parts_holder: PartsHolder, value: Variadi
 
             value.items.append(new_item)
 
+    def _deserialize_counted_variadic_values(self, parts_holder: PartsHolder, value: CountedVariadicValues):
+        if value.item_creator is None:
+            raise Exception("cannot decode list: item creator is None")
+
+        self._deserialize_single_value(parts_holder, U32Value())
+
+        while not parts_holder.is_focused_beyond_last_part():
```

Comment:
I would say this should be a `for` loop from 0 to the U32 value.
Otherwise check at line 132 that the value.length is equal to the U32 value first read.

Reply:
added a check to be sure that the expected length is the same as the number of provided values.

Reply:
Maybe for loop, indeed, to handle the more exotic case of counted variadics followed by variadics.